nix/tahoe-lafs.nix

```diff
@@ -0,0 +1,57 @@
+{ fetchFromGitHub, lib
+, nettools, python
+, twisted, foolscap, nevow, zfec
+, setuptools, setuptoolsTrial, pyasn1, zope_interface
+, service-identity, pyyaml, magic-wormhole, treq, appdirs
+, beautifulsoup4, eliot, autobahn, cryptography
+, html5lib
+}:
+python.pkgs.buildPythonPackage rec {
+  version = "1.14.0.dev";
+  name = "tahoe-lafs-${version}";
+  src = lib.cleanSource ../.;
+
+  postPatch = ''
+    sed -i "src/allmydata/util/iputil.py" \
+        -es"|_linux_path = '/sbin/ifconfig'|_linux_path = '${nettools}/bin/ifconfig'|g"
+
+    # Chroots don't have /etc/hosts and /etc/resolv.conf, so work around
+    # that.
+    for i in $(find src/allmydata/test -type f)
+    do
+      sed -i "$i" -e"s/localhost/127.0.0.1/g"
+    done
+
+    # Some tests are flaky or fail to skip when dependencies are missing.
+    # This list is over-zealous because it's more work to disable individual
+    # tests with in a module.
+    rm src/allmydata/test/test_system.py
```

Comment:
Of note here (not asking for changes in this PR): this whole suite causes pypy problems, too -- and arguably are "integration tests" anyway, so probably anything covered in test_system that's not in integration/test_*.py could be moved over (and the above deleted).

Reply:
Yep!  We should probably have a ticket: "Rewrite all of test_system.py".

Also not really directly relevant here but ... Similar to how any integration behaviors not covered by `integration/test_*.py` should have tests added there, any *unit* behaviors covered only by test_system.py and not elsewhere should similarly have *unit* tests added.

I suppose that figuring out exactly what *unit* behavior is uniquely covered by test_system.py may be somewhat possible (measure coverage from whole test suite, measure coverage from the whole test suite minus test_system.py, subtract the latter from the former?), we have no tools for measuring integration coverage so someone will actually have to understand what test_system is trying to cover to figure out what should be rewritten into `integration/test_*.py`. :/